gamla/graph.py

```diff
@@ -8,29 +8,63 @@
 from gamla import functional, functional_generic
 
 
+@toolz.curry
+def _is_better_node(node: Any, seen: Dict, key: Callable):
+    old_node = seen[key(node)]
+    return node > old_node
+
+
+@toolz.curry
+def _save_to_dict(element: Any, dictionary: Dict, func: Callable):
+    dictionary[func(element)] = element
+
+
+@toolz.curry
+def _accumulate(
+    element: Any, is_seen: Callable, better_node: Callable, keep_best: bool
+):
+    return not is_seen(element) or (keep_best and better_node(element))
+
+
 @toolz.curry
 def graph_traverse(
-    source: Any, get_neighbors: Callable, key: Callable = toolz.identity
+    source: Any,
+    get_neighbors: Callable,
+    key: Callable = toolz.identity,
+    keep_best: bool = False,
```

Comment:
don't change the signature, to avoid breaking other targets (in general default variables are bug prone and should be avoided by wrapping with functions that give these values)

Reply:
But at some point I need to add a parameter (here I added a boolean) to decide whether we do the regular usage or the new one?

Reply:
you can have one general function that exposes everything, and keep `graph_traverse` as it works today.

The `keep_best` and sort related logic is not general, so might be better placed in the user code rather than the lib.

Reply:
What do you mean by the user code? So far the only thing in the lib is the bool `keep_best`.
You mean creating a more general function where `graph_traverse` is a particular case of the latter with a specific parameter?

Reply:
I did something like that for `graph_traverse_many`